PhotoStream/app/src/main/java/com/example/siyer/photostream/CollectionFragment.java

```diff
@@ -0,0 +1,140 @@
+package com.example.siyer.photostream;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import android.widget.Toast;
+import java.util.ArrayList;
+
+/**
+ * Created by siyer on 9/28/2015.
+ * This Collection Fragment displays all the saved images in the
+ * database and allows to browse/remove them.
+ */
+
+public class CollectionFragment extends Fragment {
+    public WebView webFeedView;
+    public int feedIndex = 0; //Current Image index for WebView
+    public ArrayList<String> collectionUrl; //Array with all the URLs in my collection
+    CollDbHelper mDbHelper; //Database Helper
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        final View rootview = inflater.inflate(R.layout.collection_fragment, container, false);
+
+        mDbHelper = new CollDbHelper(getActivity()); //Initializing database and reading it
+        collectionUrl = mDbHelper.readDatabase();
+
+        webFeedView = (WebView) rootview.findViewById(R.id.webView_fd);
+        webFeedView.loadUrl(getString((R.string.default_img))); //Init WebView and load default
+        Toast.makeText(getActivity().getBaseContext(), getString(R.string.next_prev),Toast.LENGTH_SHORT).show();
+        getImageFromColl(collectionUrl);
+
+        //OnCLick Listener for NEXT button
+        final View next_but = rootview.findViewById(R.id.next_button_fd);
+        next_but.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // Various normal and edge cases
+                if (collectionUrl.size()>=1) {
+                    if (feedIndex < collectionUrl.size() - 1) {
+                        feedIndex += 1;
+                        getImageFromColl(collectionUrl);
+                    } else {
+                        feedIndex = 0;
+                        getImageFromColl(collectionUrl);
+                    }
+                }else if (collectionUrl.size()==0){
+                    Toast.makeText(getActivity().getBaseContext(), getString(R.string.no_img_toast),Toast.LENGTH_LONG).show();
+                    webFeedView.loadUrl(getString(R.string.default_img));
+                }
+            }
+        });
```

Comment:
Moving large onclicklisteners such as this to a separate method will increase clarity and re usability.